src/main/proto/th2_grpc_data_provider/data_provider.proto

```diff
@@ -121,154 +104,93 @@ message EventData {
 message EventMetadata {
   EventID event_id = 1; // Current event id
   EventID parent_event_id = 2; // Event id of a parent event. It is absent for root event
 
-  google.protobuf.Timestamp start_timestamp = 4;
-  EventStatus status = 5; // Aggregated status of current and children events which sync written.
-  string event_name = 6;
-  string event_type = 7;
+  google.protobuf.Timestamp start_timestamp = 3;
+  EventStatus status = 4; // Aggregated status of current and children events which sync written.
+  string event_name = 5;
+  string event_type = 6;
 }
 
-/* List of event ids */
-
-message EventIds {
-  repeated EventID ids = 1;
+message BulkEventRequest {
+  repeated EventID id = 1;
 }
 
-/* List of event data */
-
-message Events {
-  repeated EventData events = 1;
+message BulkEventResponse {
+  repeated EventResponse event = 1;
 }
 
-/* Parsed message info */
-
-message MessageBodyWrapper {
+message MessageGroupItem {
   Message message = 1;
   bool match = 2;
 }
 
-/* Full event data */
-
-message MessageData {
+message MessageGroupResponse {
   MessageID message_id = 1; // Message id should be unique in session
   google.protobuf.Timestamp timestamp = 2; // Message creation timestamp
   bytes body_raw = 3;
-  repeated EventID attached_event_ids = 4; // id of events depending on this message
-  repeated MessageBodyWrapper messages = 5; // May be absent.
+  repeated EventID attached_event_id = 4; // ids of events linked to this message group
+  repeated MessageGroupItem message_item = 5; // May be absent.
 }
 
-/* Message and last scanned object. With a certain frequency (this can be configured in the configuration options),
-
- * it is sent along with the elements found when searching via searchMessages */
-
-
-message LastScannedObjectInfo {
-  oneof id { // Id of last scanned message or event
-    MessageID message_id = 1;
-    EventID event_id = 2;
-  }
-  google.protobuf.Timestamp timestamp = 3; // Timestamp of message or event
-  int64 scan_counter = 4; // Number of objects already scanned
-}
-
-/* List of strings */
-
-message StringList {
-  repeated string list_string = 1;
-}
-
-/* Response of match request */
-
-message IsMatched {
-  bool is_matched = 1;
-}
-
-/* Filtering data types */
-
-enum FilterParameterType {
-  NUMBER = 0;
-  BOOLEAN = 1;
-  STRING = 2;
-  STRING_LIST = 3;
+message FilterInfoRequest {
+  FilterName filter_name = 1;
 }
 
 message FilterName {
-  string filter_name = 1;
+  string name = 1;
 }
 
-message ListFilterName {
-  repeated FilterName filter_names = 1;
+message FilterNamesResponse {
+  repeated FilterName filter_name = 1;
 }
 
-message Parameter {
+message FilterParameter {
   string name = 1;
   FilterParameterType type = 2;
   google.protobuf.Any default_value = 3;
   string hint = 4;
 }
 
-
-/* Complete information about a specific filter */
-
-message FilterInfo {
-  FilterName name = 1;
-  string hint = 2;
-  repeated Parameter parameters = 3;
+enum FilterParameterType {
+  NUMBER = 0;

```

Comment:
Probably, we should not start ids from 0 because it is a default value for enum

Reply:
Shouldn't proto3 contain a null element?
https://developers.google.com/protocol-buffers/docs/proto3#enum
I get an exception:  "The first enum value must be zero in proto3."

Reply:
Yep, my bad. You are right. There must be an element with 0 indexe in the enum. But still, it will be used as a default value. If we don't need that we should introduce a value by ourselfs